firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -207,17 +207,53 @@ public void finalizeReports(String currentSessionId) {
             TRANSFORM.reportToJson(report.withEvents(ImmutableList.from(events))));
       }
       recursiveDelete(sessionDirectory);
+      capFinalizedReports();
     }
   }
 
-  public List<CrashlyticsReport> loadFinalizedReports() {
-    final List<CrashlyticsReport> allReports = new ArrayList<>();
-    final List<File> priorityReports = getAllFilesInDirectory(priorityReportsDirectory);
-    for (File reportFile : priorityReports) {
-      allReports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
+  private void capFinalizedReports() {
+    List<File> allReportFiles = getAllFinalizedReportFiles();
+    if (allReportFiles.size() > defaultMaxReportsToKeep) {
+      // Make a sublist of the reports that go over the size limit
+      List<File> filesToRemove =
+          allReportFiles.subList(defaultMaxReportsToKeep, allReportFiles.size());
+      for (File reportFile : filesToRemove) {
+        reportFile.delete();
+      }
     }
-    final List<File> reports = getAllFilesInDirectory(reportsDirectory);
-    for (File reportFile : reports) {
+  }
+
+  @NonNull
+  private List<File> getAllFinalizedReportFiles() {
+    final ArrayList<File> allReportsFiles = new ArrayList<>();
+    List<File> priorityReports = getAllFilesInDirectory(priorityReportsDirectory);
+    Collections.sort(priorityReports, SESSION_ID_COMPARATOR);
```

Comment:
How would you feel about reducing the repeated code in these two methods by adding another helper function like so?

    private static List<File> getSortedReportFiles(List<File> priorityReports, List<File> reports) {
        Collections.sort(priorityReports, SESSION_ID_COMPARATOR);
        Collections.sort(reports, SESSION_ID_COMPARATOR);
        final ArrayList<File> allReportsFiles = new ArrayList<>();
        allReportsFiles.ensureCapacity(priorityReports.size() + reports.size());
        allReportsFiles.addAll(priorityReports);
        allReportsFiles.addAll(reports);
        return allReportsFiles;
    }

Reply:
love it